src/SimpleThings/EntityAudit/AuditReader.php

```diff
@@ -819,4 +819,39 @@ public function getEntityHistory($className, $id)
 
         return $result;
     }
+    
+    public function matchRevisions($class, $entityId, RevisionCriteria $criteria)
+    {
+        $revisions = $this->findRevisions($class, $entityId);
+        return array_filter($revisions, function($revision) use ($criteria) {
+            return $this->matchCriteria($revision, $criteria);
+        });
+    }
+
+    protected function matchCriteria(Revision $revision, RevisionCriteria $criteria)
+    {
+        return ($this->matchDates($revision, $criteria) && $this->matchUsername($revision, $criteria));
+    }
+
+    protected function matchDates(Revision $revision, RevisionCriteria $criteria)
+    {
+        if (empty($criteria->dateFrom) && empty($criteria->dateTo)) {
```

Comment:
If you like to have type safety, you will have to use getters/setters for those fields, with the appropriate type hints.